### DISCUSSION TITLE
Shared rustc compiled objects?

### DISCUSSION THREAD

**Discussion by FilipeD** [tauri-apps/tauri#12054] [Q&A] [answered]
Hello,
sorry if this has been resolved already (search returned no results) but I've found that tauri apps created both with npm and cargo GENERATE a common set of nearly 5GB worth of objects in src-tauri/target/debug/deps and src-tauri/target/debug/build (having the same filename, i.e anyhow-37710e235502b5ea). Shouldn't these be cached in a separate target folder to be used in later/similar projects?
I say separate because I noticed the project binaries are also generated in the 'build' folder...

**Comment by FabianLars** [comment, marked as answer]
First of all, your question isn't Tauri specific, everything talked about is "just" Rust.

the files in debug/deps are build artifacts of the dependencies.
the files in debug/build are the build artifacts of the build scripts of the dependencies that do have build scripts.

> Shouldn't these be cached in a separate target folder to be used in later/similar projects?

You can have a shared target dir by configuring `build.target-dir` in the cargo _config_, using the `--target-dir` cli flag, or the `CARGO_TARGET_DIR` env var. While this is generally fine to do, you can't have concurrent builds (of multiple projects) this way. It may also result in more recompilation than desired sometimes.

Alternatively you can look into `sccache`, a build cache that wouldn't require a shared target dir. 

sccache is more for saving time than disk space though.

> I say separate because I noticed the project binaries are also generated in the 'build' folder...

I only see the build script executables, not the app bins themselves.

**Reply by FabianLars** [reply]
Here's everything i've said but "official": https://doc.rust-lang.org/cargo/reference/build-cache.html

**Reply by FilipeD** [reply]
Thanks for taking the time, happy new year to you all!
